### ISSUE TITLE
Serenity throws ClassCastException on a generic method

### ISSUE THREAD

**Issue by mtedone** [serenity-bdd/serenity-core#71]
Hi, I've got the following step which works fine for all tests except this one: 

```
@Step("When I invoke the factory manufacturing for {0}")
    public <T> T whenIInvokeTheFactoryForClass(Class<T> className, PodamFactory podamFactory) throws Exception {
        return podamFactory.manufacturePojo(className);
    }
```

And the following JUnit test: 

```
@Test(expected=PodamMockeryException.class)
    @Title("Podam should throw an exception if a POJO contains invalid getters or setters")
    public void podamShouldThrowAnExceptionIfAPojoContainsInvalidGettersOrSetters() throws Exception {

        PodamFactory podamFactory = podamFactorySteps.givenAStandardPodamFactory();
        BadlyTypedPojo pojo = podamFactory.manufacturePojo(BadlyTypedPojo.class);
        Assert.assertNotNull("Manufacturing failed", pojo);

    }
```

The above passes. 

However, if I use Serenity steps, like so: 

```
 @Test(expected=PodamMockeryException.class)
    @Title("Podam should throw an exception if a POJO contains invalid getters or setters")
    public void podamShouldThrowAnExceptionIfAPojoContainsInvalidGettersOrSetters() throws Exception {

        PodamFactory podamFactory = podamFactorySteps.givenAStandardPodamFactory();
        BadlyTypedPojo pojo = podamInvocationSteps.whenIInvokeTheFactoryForClass(BadlyTypedPojo.class, podamFactory);


    }
```

I get the following ClassCastException: 

```
java.lang.Exception: Unexpected exception, expected<uk.co.jemos.podam.exceptions.PodamMockeryException> but was<java.lang.ClassCastException>
	at org.junit.internal.runners.statements.ExpectException.evaluate(ExpectException.java:28)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70)
	at net.serenitybdd.junit.runners.SerenityRunner.runChild(SerenityRunner.java:435)
	at net.serenitybdd.junit.runners.SerenityRunner.runChild(SerenityRunner.java:59)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
	at net.serenitybdd.junit.runners.SerenityRunner.run(SerenityRunner.java:248)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:160)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:78)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:212)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:68)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:140)
Caused by: java.lang.ClassCastException: uk.co.jemos.podam.test.unit.steps.PodamInvocationSteps$$EnhancerByCGLIB$$625e929c cannot be cast to uk.co.jemos.podam.test.dto.BadlyTypedPojo
	at uk.co.jemos.podam.test.unit.features.edgeCases.AbnormalPojosTest.podamShouldThrowAnExceptionIfAPojoContainsInvalidGettersOrSetters(AbnormalPojosTest.java:136)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at net.serenitybdd.junit.runners.SerenityStatement.evaluate(SerenityStatement.java:24)
	at org.junit.internal.runners.statements.ExpectException.evaluate(ExpectException.java:19)
	... 20 more
```

**Comment by wakaleo** [comment]
Does the PodamMockeryException class have a constructor with a String or Object as a parameter? Serenity (currently) needs this to restore the exceptions correctly.

**Comment by mtedone** [comment]
It does now but I still get the same error. Before it had a constructor with String, Throwable

**Comment by wakaleo** [comment]
Should be fixed in 1.0.53, as long as the custom assertion class has a constructor derived from one of the Throwable constructors.
